src/gui/SampleWaveform.cpp

```diff
@@ -28,41 +28,50 @@ namespace lmms::gui {
 
 void SampleWaveform::visualize(Parameters parameters, QPainter& painter, const QRect& rect)
 {
-	const auto x = rect.x();
-	const auto height = rect.height();
-	const auto width = rect.width();
-	const auto centerY = rect.center().y();
+	const int x = rect.x();
+	const int height = rect.height();
+	const int width = rect.width();
+	const int centerY = rect.center().y();
 
-	const auto halfHeight = height / 2;
+	const int halfHeight = height / 2;
 
 	const auto color = painter.pen().color();
 	const auto rmsColor = color.lighter(123);
 
-	const auto framesPerPixel = std::max<size_t>(1, parameters.size / width);
+	const float framesPerPixel = std::max<float>(1.0f, static_cast<float>(parameters.size) / width);
 
-	constexpr auto maxFramesPerPixel = 512;
-	const auto resolution = std::max<size_t>(1, framesPerPixel / maxFramesPerPixel);
-	const auto framesPerResolution = framesPerPixel / resolution;
+	constexpr auto maxFramesPerPixel = 512.0f;
+	const float resolution = std::max<float>(1.0f, static_cast<float>(framesPerPixel / maxFramesPerPixel));
+	const float framesPerResolution = framesPerPixel / resolution;
 
-	const auto numPixels = std::min<size_t>(parameters.size, width);
+	const size_t numPixels = std::min<size_t>(parameters.size, width);
 	auto min = std::vector<float>(numPixels, 1);
 	auto max = std::vector<float>(numPixels, -1);
-	auto squared = std::vector<float>(numPixels);
+	auto squared = std::vector<float>(numPixels, 0);
+
+	const size_t maxFrames = numPixels * static_cast<size_t>(framesPerPixel);
+
+	int pixelIndex = 0;
 
-	const auto maxFrames = numPixels * framesPerPixel;
 	for (int i = 0; i < maxFrames; i += resolution)
 	{
-		const auto pixelIndex = i / framesPerPixel;
-		const auto frameIndex = !parameters.reversed ? i : maxFrames - i;
+		pixelIndex = static_cast<int>(i / framesPerPixel);
```

Comment:
We can remove `static_cast<int>` since `i / framesPerPixel` is implicitly converted to int.